### PR TITLE
[kube-prometheus-stack] Add template to storage configurations

### PR DESCRIPTION
**Pull request by nhudson**
Signed-off-by: Nick Hudson <nick.hudson@gmail.com>

#### What this PR does / why we need it

Adds the `tpl` function around the `storageSpec` configuration in prometheus templates and `storage` configuration in the alertmanager templates.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 39.10.0
+version: 39.11.0
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:25]:
```suggestion
version: 40.1.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 1 above)
Comment (nhudson) [reply charts/kube-prometheus-stack/Chart.yaml]: Thanks.  Updated

File: charts/kube-prometheus-stack/Chart.yaml [hunk 2]
```diff
@@ -22,7 +22,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 40.0.0
+version: 40.1.0
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:25]:
```suggestion
version: 40.0.1
```

Comment (monotek) [reply charts/kube-prometheus-stack/Chart.yaml:25]: sorry, typo...

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 2 above)
Comment (nhudson) [reply charts/kube-prometheus-stack/Chart.yaml]: ha no worries.  Fixed!

File: charts/kube-prometheus-stack/Chart.yaml [hunk 3]
```diff
@@ -22,7 +22,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 40.0.0
+version: 40.0.1
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:25]:
```suggestion
version: 40.0.2
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**